theano/gof/op.py

```diff
@@ -1377,7 +1436,12 @@ def get_op_params(self):
         The names must be strings that are not a C keyword and the
         values must be strings of literal C representations.
 
+        If your COp defines an attribute `enum` created from ``enum_list()``
+        or ``enum_type()``, then this method will try to deduce macros from
+        the enum fields.
         """
+        if hasattr(self, 'enum') and hasattr(self.enum, '__dict__'):
+            return [(k, str(v)) for (k, v) in self.enum.__dict__.items()]
```

Comment:
This will break in horrible and interesting ways since a \_\_dict__ can contain all sorts of things and we don't check the type of the field.

Also, this will also break for any op that overrides get_op_params() like they are supposed to do.

Finally get_op_params() is a horrible name and is/was a temporary stopgap measure before we could pass multiple things with the params argument.  Now that we can, we should phase it out.